hdmf_ml/results_table.py

```diff
@@ -0,0 +1,98 @@
+from hdmf.utils import docval, popargs
+from hdmf.common import get_class, register_class, DynamicTable
+from hdmf.container import Container
+import numpy as np
+from sklearn.preprocessing import LabelEncoder
+
+
+@register_class('ResultsTable', 'hdmf-ml')
+class ResultsTable(get_class('ResultsTable', 'hdmf-ml')):
+
+    @docval({'name': 'cls',         'type': (str, type), 'doc': 'data for this column'},
+            {'name': 'data',        'type': 'array_data', 'doc': 'data for this column'},
+            {'name': 'name',        'type': str,     'doc': 'the name of this column'},
+            {'name': 'description', 'type': str,     'doc': 'a description for this column'},
+            allow_extra=True)
+    def __add_col(self, **kwargs):
+        cls, data, name, description = popargs('cls', 'data', 'name', 'description', kwargs)
+        if len(self.id) < len(data):
+            self.id.extend(np.arange(len(self.id), len(data)))
```

Comment:
1. Doesn't add_column already cover this?
2. This is dangerous if the table already has 2 columns and 2 rows, and data has 3 rows, then this would make the ID column longer than the existing columns.

Reply:
If I don't do that, get this error:
ValueError: column must have the same number of rows as 'id'

There is no logic in DynamicTable for automatically setting `id` when you're adding whole columns--only when you're adding row by row.

Reply:
I don't understand the use case. Will this be used only when `len(self.id) == 0`? Otherwise, see point 2 above.

Reply:
The use case is I just want to add whole columns to ResultsTable. It will only be used when `len(self.id) == 0`. In that case, `add_column` does not handle that.

Reply:
I updated that method to be more specific and raise an error if sizes don't match up.